src/app/shared/filter-bar/label-filter-bar/label-filter-bar.component.html

```diff
@@ -23,7 +23,7 @@
           <mat-list-option
             #option
             *ngFor="let label of this.labels$ | async"
-            [value]="label.name"
+            [value]="label.formattedName"
             [selected]="selectedLabelNames.includes(label.name)"
```

Comment:
What about the other uses of label.name instead of formattedName, will there be any edge cases from changing only the value here

Reply:
This `value` being changed to `label.formattedName` only affects the values in `LabelFilterBarComponent::selectedLabelNames`. It is only used internally, to update the value of the behaviour subject field `LabelFilterBarComponent::selectedLabels`. In `LabelFilterBarComponent::simulateClick` and `LabelFilterBar::updateSelection` methods:

```ts
this.selectedLabels.next(this.selectedLabelNames);
```

`LabelFilterBarComponent::updateSelection` value is an `@Input` field. When looking for places that this component is used, it is only used in `FilterBarComponent`, in `filter-bar.component.html`:

```html
<mat-grid-tile class="label-filter-grid-tile" colspan="1">
  <app-label-filter-bar [selectedLabels]="this.labelFilter$" [hiddenLabels]="this.hiddenLabels$"></app-label-filter-bar>
</mat-grid-tile>
```

Hence any changes to `LabelFilterBarComponent::selectedLabels` will only cause a change to `FilterBarComponent::labelFilter$`. Looking at the places that this field is used, it is only used in `FilterBarComponent::ngAfterViewInit`, which subscribes to changes to this field:

```ts
    this.labelFilterSubscription = this.labelFilter$.subscribe((labels) => {
      this.dropdownFilter.labels = labels;
      this.applyDropdownFilter();
    });
```

On every change, the `FilterBarComponent::dropdownFilter.labels` is changed, and then `applyDropdownFilter` is called, where the filter in each of the `views$` are updated to `FilterBarComponent::dropdownFilter`. Here, the `views$` are the child components that are filterable. It is an `@Input` field. Looking at places where `FilterBarComponent` is used, only in `issue-viewer.component.html`:

```html
<app-filter-bar [views$]="views" #filterbar></app-filter-bar>
```

The value of `FilterBarComponent::views$` takes the value of `IssueViewer::views`. `IssueViewer::views` is a field that follows the value of `IssueViewer::cardViews`:

```ts
  ngAfterViewInit(): void {
    this.viewChange = this.cardViews.changes.subscribe((x) => this.views.next(x));
  }
```

Which in turn is a `@ViewChildren` component that is of type `QueryList<CardViewComponent>`! This means that any change to `FilterBarComponent::dropdownFilter` will cause a change to the filters of all `CardViewComponent`s within `IssueViewer`.

TLDR, in short, the change of `value` from `label.name` to `label.formattedName` only means that all `CardViewComponent`s filter by `label.formattedName` instead of `label.name`. Note that `CardViewComponent` has selector `app-view-card`, so to double check where this change will affect, paste the following code to the browser's console when opening the app:

```js
document.querySelectorAll('app-card-view')
```

We see that the `CardViewComponent`s are only the columns of the contributors of the current repo. This is the only component that takes the intended effect of the change.

I realised that the `LabelFilterBarComponent::hiddenLabelNames` needs a similar change. Similar to what I have explained above, the `CardViewComponent`s are the only component that takes the intended effect of the change. However, looks like it only hide the labels. I'm not sure if this is the intended behaviour of the hiding feature, if it not, there needs to be a change to how the `CardViewComponent`s make use of the `filter.hiddenLabels`.

![image](https://github.com/CATcher-org/WATcher/assets/87511888/fa58653b-b92b-4c74-882e-d4bbba5b6f61)

After hiding the label `difficulty.Easy`:

![image](https://github.com/CATcher-org/WATcher/assets/87511888/a5017094-6b5c-4264-9c57-1733bb6c25ee)

Reply:
> However, looks like it only hide the labels. I'm not sure if this is the intended behaviour of the hiding feature, if it not, there needs to be a change to how the CardViewComponents make use of the filter.hiddenLabels.

To confirm, what did you think the hiding feature should do? We might want to reconsider how we present the features or change the feature if it doesn't make intuitive sense to the user

Reply:
I think it should hide the issues & PRs that have the tag, instead of simply hide the tag from the page. I have posted a new issue on this at #240.